src/pages/CartPage.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect } from "react";
+import { Box, Typography, Button,  IconButton, Paper } from "@mui/material";
+import { useCart } from "../context/cardContext";
+import DeleteIcon from "@mui/icons-material/Delete";
+import { styled } from "@mui/system";
+import { useNavigate } from "react-router-dom";
+import Layout from "../Layout/layout";
+
+
+const CartItemContainer = styled(Paper)(({ theme }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  padding: theme.spacing(2),
+  borderRadius: theme.spacing(1),
+  marginBottom: theme.spacing(2),
+}));
+
+const CartItemDetails = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  alignItems: 'center',
+}));
+
+const CartItemImage = styled('img')({
+  width: 100,
+  height: 100,
+  marginRight: 20,
+});
+
+const CartSummary = styled(Paper)(({ theme }) => ({
+  padding: theme.spacing(2),
+  borderRadius: theme.spacing(1),
+  marginTop: theme.spacing(2),
+}));
+
+const CartPage: React.FC = () => {
+  const { cartItems, removeFromCart, updateQuantity, clearCart } = useCart();
+
+  const handleRemoveFromCart = (id: string) => {
+    removeFromCart(id);
+  };
+
+  const handleQuantityChange = (id: string, quantity: number) => {
+    updateQuantity(id, quantity);
+  };
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (cartItems.length === 0) {
+      const timer = setTimeout(() => {
+        navigate("/products");
+      }, 3000); 
+      return () => clearTimeout(timer); 
+    }
+  }, [cartItems, navigate]);
```

Comment:
Why do you need to navigate user back if cart is empty? Websites usually show a message i.e.`Your cart is empty.` with a button `Back to shopping`